### PR TITLE
feat(jax): support D3 parameters under jit

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit-Ops Pull Request

## Description

Register `D3Parameters` as a JAX PyTree so callers can pass the existing dataclass directly to JIT-compiled DFT-D3 calculations.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Register DFT-D3 parameter arrays as dynamic PyTree data and the interpolation mesh as static metadata.
- Pass `D3Parameters` directly through compiled matrix, CSR, zero-edge, and empty-system calls.
- Update the JAX DFT-D3 example to remove manual parameter unpacking and reconstruction.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

Baset on #158. 

None.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -204,7 +204,7 @@ jobs:
     uses: ./.github/workflows/reusable-gpu-tests.yml
     with:
       suite: impacted
-      timeout-minutes: 120
+      timeout-minutes: 180
```
Comment (M-R-Schaefer) [inline .github/workflows/ci.yml]: Did the warp migration or D3 changes introduce any slowdowns?

File: nvalchemiops/jax/interactions/dispersion/_dftd3.py [hunk 2]
```diff
@@ -1102,6 +1102,13 @@ def max_z(self) -> int:
         return self.rcov.shape[0] - 1
 
 
+jax.tree_util.register_dataclass(
+    D3Parameters,
+    data_fields=["rcov", "r4r2", "c6ab", "cn_ref"],
+    meta_fields=["interp_mesh"],
+)
```
Comment (M-R-Schaefer) [inline nvalchemiops/jax/interactions/dispersion/_dftd3.py:1109]: Minor nitpick: One could turn this into a decorator so the dataclass registration is located directly at the class definition.

```python 
@partial(
    jax.tree_util.register_dataclass,
    data_fields=["rcov", "r4r2", "c6ab", "cn_ref"],
    meta_fields=["interp_mesh"],
)
@dataclass
class D3Parameters:
    ...
```

Optional

**Review by M-R-Schaefer — CHANGES_REQUESTED** [review]
Looks good. The migration is clean. Two minor comments, The decorator related one is definitely optional.

**Review by M-R-Schaefer — DISMISSED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zubatyuk) [reply .github/workflows/ci.yml]: Fighting CI timeouts

File: nvalchemiops/jax/interactions/dispersion/_dftd3.py (hunk 2 above)
Comment (zubatyuk) [reply nvalchemiops/jax/interactions/dispersion/_dftd3.py]: Fixed in 45b7e11.

**Review by dallasfoster — APPROVED** [review]

**Comment by dallasfoster** [comment]
/ok to test [9874aa7](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/162/commits/9874aa7bd660edad2ac674176dcc37b7bdfc20a6)
